src/System.Private.CoreLib/shared/System/TimeSpan.cs

```diff
@@ -239,11 +239,10 @@ private static TimeSpan Interval(double value, int scale)
         {
```

Comment:
It may be better to pass in `scale` as double while you are on it. Converting `int` to `double` is not free.

Reply:
shouldn't it be long ? as all ticksPerDay, ticksPerMinutes are long ?

Reply:
The scale constant needs to be converted to double before multiplication. Converting int to double vs. long to double is unlikely to make a difference. Avoid the conversion completely is better. (You can try to measure it.)